org.eclipse.xtext.xbase.ui/src/org/eclipse/xtext/xbase/ui/hover/JvmAnnotationReferencePrinter.java

```diff
@@ -0,0 +1,220 @@
+/**
+ * Copyright (c) 2015, 2020 itemis AG (http://www.itemis.eu) and others.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.xtext.xbase.ui.hover;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.eclipse.emf.common.util.EList;
+import org.eclipse.emf.common.util.URI;
+import org.eclipse.emf.ecore.EStructuralFeature;
+import org.eclipse.emf.ecore.util.EcoreUtil;
+import org.eclipse.xtext.common.types.JvmAnnotationReference;
+import org.eclipse.xtext.common.types.JvmAnnotationType;
+import org.eclipse.xtext.common.types.JvmAnnotationValue;
+import org.eclipse.xtext.common.types.JvmOperation;
+import org.eclipse.xtext.ui.editor.hover.html.XtextElementLinks;
+import org.eclipse.xtext.xbase.XAbstractFeatureCall;
+import org.eclipse.xtext.xbase.XBinaryOperation;
+import org.eclipse.xtext.xbase.XBooleanLiteral;
+import org.eclipse.xtext.xbase.XFeatureCall;
+import org.eclipse.xtext.xbase.XListLiteral;
+import org.eclipse.xtext.xbase.XMemberFeatureCall;
+import org.eclipse.xtext.xbase.XNumberLiteral;
+import org.eclipse.xtext.xbase.XStringLiteral;
+import org.eclipse.xtext.xbase.XTypeLiteral;
+import org.eclipse.xtext.xbase.annotations.xAnnotations.XAnnotation;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Iterables;
+
+/**
+ * Provides a HTML representation for {@code JvmAnnotationReference}s to be used in the hover.
+ * 
+ * @author Sven Efftinge - Initial contribution and API
+ */
+public class JvmAnnotationReferencePrinter {
+
+	/**
+	 * @return the HTML representation for the passed {@code reference}
+	 * @throws NullPointerException
+	 *             if the {@code reference} is {@code null}
+	 */
+	public String toHtmlString(JvmAnnotationReference reference) {
+		Preconditions.checkNotNull(reference);
+		return internalToString(reference);
```

Comment:
The original code had exception-handling code here. I chose to remove the same as it was not required.

Reply:
The runtime exception was most likely caught to avoid NPEs or IllegalArgumentException. Not sure if we have a tradeoff between robustness and clean code here.

Reply:
Actually, after reading your comment, I chose to revert back the changes I made to this method.  I did keep `Preconditions.checkNotNull(reference)` just to be on the safer side.